BitwardenShared/Core/Platform/Services/MigrationService.swift

```diff
@@ -178,39 +171,38 @@ class DefaultMigrationService {
     /// Performs migration 3.
     ///
     /// Notes:
-    ///   - Migrates the integrity state values for the extensions to the new key values.
+    ///   - Previously this migrated the MAUI integrity state values to the native key format.
+    ///     Since that's been removed, this just removes the MAUI integrity state values.
     ///
     private func performMigration3() async throws {
-        defer {
-            // Remove the previous values.
-            appGroupUserDefaults.removeObject(forKey: "bwPreferencesStorage:iOSAutoFillBiometricIntegritySource")
-            appGroupUserDefaults.removeObject(forKey: "bwPreferencesStorage:iOSExtensionBiometricIntegritySource")
-            appGroupUserDefaults.removeObject(forKey: "bwPreferencesStorage:iOSShareExtensionBiometricIntegritySource")
-        }
+        appGroupUserDefaults.removeObject(forKey: "bwPreferencesStorage:biometricIntegritySource")
+        appGroupUserDefaults.removeObject(forKey: "bwPreferencesStorage:iOSAutoFillBiometricIntegritySource")
+        appGroupUserDefaults.removeObject(forKey: "bwPreferencesStorage:iOSExtensionBiometricIntegritySource")
+        appGroupUserDefaults.removeObject(forKey: "bwPreferencesStorage:iOSShareExtensionBiometricIntegritySource")
+    }
 
+    /// Performs migration 4.
+    ///
+    /// Notes:
+    ///   - Removes the integrity state values.
+    ///
+    private func performMigration4() async throws {
```

Comment:
🤔 Thinking through the situation where we might need to revert this change and re-add the integrity in the future, such as if there's some other implication that we're not realizing right now that means we need it back. This would mean that a user using the app with re-added integrity would just need to turn Face ID (etc.) off and then back on again with both the app and the extension(s) to get it working again, correct?

That's not great, but doesn't seem too particularly onerous, I just want to make sure there's nothing I'm missing.

Reply:
Yeah, either turn it off and back on, or we could potentially get the integrity state value on the next successful Face ID scan and persist that.

Reply:
Seems reasonable. Not necessarily super thrilling to tell all our users to do that, but at least doable.